src/base/b-dynamic-page/b-dynamic-page.ts

```diff
@@ -477,14 +483,26 @@ export default class bDynamicPage extends iDynamicPage {
 				}
 
 				let
-					newPage = e;
+					newPageInfo = e;
 
 				if (Object.isTruly(this.eventConverter)) {
-					newPage = Array.concat([], this.eventConverter).reduce((res, fn) => fn.call(this, res, this.page), newPage);
+					newPageInfo = Array
+						.concat([], this.eventConverter)
+						.reduce((res, fn) => fn.call(this, res, this.page), newPageInfo);
 				}
 
-				if (newPage == null || Object.isString(newPage)) {
-					this.page = <string>newPage;
+				const
+					[newPageComponentName, newPageKey = newPageComponentName] = newPageInfo ?? [],
+					pageChanged = newPageComponentName !== this.page,
+					oldPageKey = this.pageKey;
+
+				if (newPageComponentName == null || Object.isString(newPageComponentName)) {
+					this.page = newPageComponentName;
+
+					if (!pageChanged && newPageKey !== oldPageKey) {
+						this.pageKey = newPageKey;
+						this.syncPageWatcher(newPageComponentName, this.page);
```

Comment:
```suggestion
						this.syncPageWatcher(this.page, this.page);
```

или

```suggestion
						this.syncPageWatcher(newPage, newPage);
```

твоя запись по смыслу является тем же самым, но из-за разных переменных может создаться впечатление, что там разные значения

Reply:
По моему мнению, мой вариант как раз лучше. Мы натурально передаем туда то, что должно передаваться по смыслу, а тот факт, что они на самом деле одинаковые, уже вторичен. Если же передавать одну и ту же переменную, то мы искусственно "вскрываем" эту логику, и при беглом чтении, мне кажется, это только больше вопросов вызовет - почему в обработчик изменения передается одно и то же значение.

Reply:
Для меня как раз первично, что там одинаковые значения, чтобы не возникло ложного восприятия, что они разные. Насчёт беглого чтения - соглашусь, такая конструкция будет понятная только в контексте предшествующего кода.

Reply:
Нужно третье мнение

Reply:
Мне больше вариант Андрея нравится